packages/shared/src/contexts/NotificationsContext.tsx

```diff
@@ -1,39 +1,28 @@
 import React, {
   ReactElement,
   ReactNode,
+  useCallback,
   useContext,
   useEffect,
-  useMemo,
   useRef,
   useState,
 } from 'react';
-import {
-  UseNotificationPermissionPopup,
-  useNotificationPermissionPopup,
-} from '../hooks/useNotificationPermissionPopup';
-import usePersistentContext from '../hooks/usePersistentContext';
 import { BootApp } from '../lib/boot';
-import { isDevelopment, isTesting } from '../lib/constants';
-import { checkIsExtension } from '../lib/func';
-import AuthContext from './AuthContext';
 import { AnalyticsEvent, NotificationPromptSource } from '../lib/analytics';
 import { useAnalyticsContext } from './AnalyticsContext';
+import { SubscriptionCallback } from '../hooks/notifications/useOneSignal';
+import {
+  UsePushNotification,
+  usePushNotification,
+} from '../hooks/notifications/usePushNotification';
 
-export interface NotificationsContextData
```

Comment:
Simplified the returned properties by either encapsulating it through the single object or by introducing a hook if it is something that can be accessed through `useQuery`.

Reply:
I actually debated moving this to it's own context called `InApp` or something or `Push` since it's only used on webapp .
Let me know if you think that makes sense.

Reply:
Yeah, that makes sense too. Especially they seem like have nothing to do with one another.

Reply:
Yeah if it's not a "major" effort I would suggest to do it here since you already abstracted it mostly.

Reply:
Yeah, mostly copy-paste, would just require a bit of work. Let me fix the blockers first, then I will get back to this.